python/pyspark/mllib/linalg/__init__.py

```diff
@@ -1296,9 +1296,19 @@ def asML(self):
         return newlinalg.SparseMatrix(self.numRows, self.numCols, self.colPtrs, self.rowIndices,
                                       self.values, self.isTransposed)
 
-    # TODO: More efficient implementation:
     def __eq__(self, other):
-        return np.all(self.toArray() == other.toArray())
+        if ((type(other) is not type(self)) or
```

Comment:
Ok. If this is the case then the `__eq__` method of the `DenseMatrix` class is also in trouble, as it returns `False` when they're not both `DenseMatrix` classes.